core/app/models/spree/stock/allocator/availability.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Spree
+  module Stock
+    module Allocator
+      class Availability < Spree::Stock::Allocator::Base
+        private
+
+        def allocate_inventory(availability_by_location)
+          availability_by_location.transform_values do |available|
+            # Find the desired inventory which is available at this location
+            packaged = available & @desired
+
+            # Remove found inventory from desired
+            @desired -= packaged
```

Comment:
`@desired` -> `desired` and make `attr_reader -> attr_accessor` for that attribute.